apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/html/libraries/iframe-edit-mode.js.ts

```diff
@@ -425,14 +435,36 @@ function initDragAndDrop () {
                 }).catch(e => {
                     handleHttpErrors(e);
                 })
-            } else { // Adding specific Content Type
-                window.contentletEvents.next({
-                    name: 'add-content',
-                    data: {
-                        container: container.dataset,
-                        contentType: draggedContent
+            } else { // Adding specific Content Type / Contentlet
+                // Content Type
+                if (!draggedContent.contentType) {
+                    window.contentletEvents.next({
+                        name: 'add-content',
+                        data: {
+                            container: container.dataset,
+                            contentType: draggedContent
+                        }
+                    });
+                } else if (draggedContent.contentType) { // Contentlet
+
+                    if (draggedContent.contentType === 'FORM') {
+                        const requestForm = async () => {
+                            const url = 'api/v1/containers/form/' + draggedContent.id + '?containerId=' + container.dataset['dotIdentifier'];
+                            const response = await fetch(url);
+                            const json = await response.json();
+
+                            sendCreateContentletEvent({ 
+                                baseType: 'FORM',
+                                identifier: json.entity.content.identifier,
+                                inode: json.entity.content.inode
+                            });
+                        }
+                        requestForm();
```

Comment:
Don't you need an `await` here? Also, no error management here.

Reply:
no,

![image](https://user-images.githubusercontent.com/37185433/141834709-cd9c7c7f-abc6-4d22-8e42-ff6835f8d1ee.png)


that `async` is added to an encapsulated function, so I can use `await` on lines 444 and 445.

also when `requestForm()` is called, there is no extra functionality that depends on that request to be fulfilled